### DISCUSSION TITLE
Ender 3 S1 Pro support (touch screen)?

### DISCUSSION THREAD

**Discussion by Eldenroot** [mriscoc/Ender3V2S1#450] [Q&A]
What about a support for S1 PRO version? 

Thank you!

**Comment by H0Z0** [comment]
It is supported if you replace the display with DWIN or DACAI. Touch screen is not supported.

**Reply by Chihhao** [reply]
Hi @H0Z0 , what does DWIN or DACAI mean? is the S1 (not pro) 's screen suitable? or 12864?

I have a Ender 3 S1 Pro. 
If i replace the stock touch screen with S1's screen, it will support octoprint perfectly.
Without replace screen, it will also support octoprint as long as upgrade firmware to "Ender3S1-F4-20221002.bin", but screen will not show perfect info when printing.
Did I get that right?

**Reply by H0Z0** [reply]
Yes, the DWIN and DACAI are screens for S1 and E3V2 printers, not 12864. DWIN supports gcode preview, DACAI does not.
I don't have a touch screen, so I can't test whether it shows anything at all or just a blank screen. You can't use the menus normally, that's for sure. Maybe you can control it from octoprint regardless of the screen and the printing works properly. But then you lose the extra features of prof fw, which you could control from the screen.

**Reply by Chihhao** [reply]
thank you!

**Comment by Dune29527** [comment]
Is there any plan to update so that the touch screen would be supported?

**Reply by MarkforU** [reply]
Believe, and I maybe wrong, but because of the EEPROM size of the 4.2.2 and 4.2.7 there isn't room for the extra needs of touch screen or we'd probably see it by now. This isn't the first time I've seen that question but I switched all my screens on my printers to DWIN and DACAI screens and I'm using Octoprint for a print farm. Just saw this was 4mo old post, but I'm sure the answer will come out soon for you...believe me, a lot of people have asked me if they should get a touch screen when upgrading for this Pro Firmware and I tell them No! Buy the $40 screen and use the knob.

**Reply by grrmisfit** [reply]
> Is there any plan to update so that the touch screen would be supported?

until creality releases source code for the touch screen no custom firmware can support it. Buy an ender v2 screen and run pro firmware or bug creality to release source.

**Reply by Dune29527** [reply]
was hoping, but it looks like I'm buying the knob screen. thanks.

**Comment by hayden-t** [comment]
just buy the knob screen its actually better

**Comment by TheWebMachine** [comment]
If you are seeking a non-AliExpress source for the DACAI screen, I found them on Amazon for a reasonable price and confirm they work with this firmware, just be sure to reflash the screen with the proper assets as discussed in the install guide. You'll also have to remove the bottom cover of the printer base and swap the display cable, which is included with the new screen. I left the touchscreen cable hooked up, as well, so I can (hopefully) help with development of support for those screens in the future....it was also hot-glued in place and I didn't feel like picking that away. Just tuck the unused cable into the base if you want. There's plenty of room and no moving parts in there besides the cooling fan in the corner.

https://www.amazon.com/gp/product/B098WFCQYW
(clean link, no reftags or anything like that)

**Reply by Dune29527** [reply]
Very cool, thank you

**Reply by cedeix** [reply]
hello

I have the same problem ... i would like to use the mrisco firmware on my E3S1Pro with touchscreen.
my question is what about the pin assignment of the S1Pro touch display and the Ender3 V2 display with the rotary knob?
as the connections look different.
